marlowe-playground-client/src/Marlowe/Holes.purs

```diff
@@ -646,13 +632,13 @@ replaceInPositions constructor firstHole@(MarloweHole { marloweType, name }) hol
 
     holeString = constructMarloweType constructor firstHole m
 
-    stringLengthDifference = (length holeString) - (length name)
+    stringLengthDifference = (length holeString) - (length name) - 1
 
     (final /\ _) =
       foldl
         ( \(currString /\ currLength) hole@(MarloweHole { name, start, end }) ->
             let
-              newString = replaceInPosition { start, end, string: currString, replacement: holeString }
+              newString = replaceInPosition { start: start + currLength, end: end + currLength, string: currString, replacement: holeString }
```

Comment:
I think a bug had been introduced here, this is how it should be, you add the extra length of string every time you move onto the next hole, otherwise you will be replacing the wrong part of the contract string.